src/main/java/org/apache/cassandra/sidecar/MainModule.java

```diff
@@ -304,4 +314,28 @@ public ChecksumVerifier checksumVerifier(Vertx vertx)
     {
         return new MD5ChecksumVerifier(vertx.fileSystem());
     }
+
+    @Provides
+    @Singleton
+    @Named("SidecarVersion")
+    private static String sidecarVersion()
+    {
+        final String resource = "/sidecar.version";
+        try (InputStream input = NodeSettings.class.getResourceAsStream(resource);
+             ByteArrayOutputStream output = new ByteArrayOutputStream())
+        {
+            byte[] buffer = new byte[32];
+            int length;
+            while ((length = input.read(buffer)) >= 0)
+            {
+                output.write(buffer, 0, length);
+            }
+            return output.toString(StandardCharsets.UTF_8.name());
+        }
+        catch (Exception exception)
+        {
+            LOGGER.error("Failed to retrieve Sidecar version", exception);
+        }
+        return "unknown";
+    }
```

Comment:
There is no test for the method (reading from the resource stream). Can you add a test?

Reply:
This is a tricky one — since the version number is embedded into the JAR itself, I don't see what to compare it with. I could read it out from resources in the test code as well, but then I'll be just testing two copies of the same code against each other, which may not be helpful.

Reply:
Well, the reason I think value makes more sense is because it allows you to potentially retry reading from the file if retrieving the version fails during `MainModule` initialization.

Reply:
Regarding the test, you can add a *test* resource called `sidecar.version`, and write a testing like this.

```java
    @Test
    public void testReadSidecarVersion()
    {
        String version = injector.getInstance(Key.get(String.class, Names.named("SidecarVersion")));
        Assertions.assertThat(version).isNotNull().isEqualTo("1.0.0");
    }
```

Reply:
Done, thank you.